agent/consul/acl.go

```diff
@@ -260,8 +219,9 @@ func (r *ACLResolver) resolveTokenLegacyAsync(token string, cached *structs.Auth
 	if err == nil {
 		parent := acl.RootAuthorizer(reply.Parent)
 		if parent == nil {
-			r.fireAsyncLegacyResult(token, cached.Authorizer, cacheTTL, acl.ErrInvalidParent)
-			return
+			// TODO(rb): what if cached is nil?
```

Comment:
I think we should do something like

```
var authz acl.Authorizer
if cached != nil {
   authz = cached.Authorizer
}
```

Then instead of `cached.Authorizer` use `authz` for the next 2 lines.

In practice `reply.Parent` is always going to be one of `allow`, `deny` or `manage` so calling `acl.RootAuthorizer` is never going to return `nil`